test/harness/e2e/harness.go

```diff
@@ -0,0 +1,108 @@
+package e2e
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/flightctl/flightctl/api/v1alpha1"
+	apiclient "github.com/flightctl/flightctl/internal/api/client"
+	client "github.com/flightctl/flightctl/internal/client"
+	"github.com/flightctl/flightctl/test/harness/e2e/vm"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+const POLLING = "250ms"
+const TIMEOUT = "60s"
+
+type Harness struct {
+	VM        vm.TestVMInterface
+	Client    *apiclient.ClientWithResponses
+	Context   context.Context
+	ctxCancel context.CancelFunc
+}
+
+func findTopLevelDir() string {
+	currentWorkDirectory, err := os.Getwd()
+	Expect(err).ToNot(HaveOccurred())
+
+	parts := strings.Split(currentWorkDirectory, "/")
+	for i := len(parts) - 1; i >= 0; i-- {
+		if parts[i] == "test" {
+			path := strings.Join(parts[:i], "/")
+			logrus.Debugf("Top-level directory: %s", path)
+			return path
+		}
+	}
+	Expect(false).To(BeTrue(), "could not find top-level directory")
```

Comment:
can you return an empty string and then validate the return to maybe not sure if this would just work
```
Fail("Could not find top-level directory")
```

Reply:
Thanks. I remember writing that and thinking... I am sure there is a better way :D